src/serac/numerics/functional/tensor.hpp

```diff
@@ -1170,13 +1172,28 @@ SERAC_HOST_DEVICE bool is_symmetric_and_positive_definite(tensor<double, 3, 3> A
 }
 
 /**
- * @brief Solves Ax = b for x using Gaussian elimination with partial pivoting
- * @param[in] A The coefficient matrix A
- * @param[in] b The righthand side vector b
- * @note @a A and @a b are by-value as they are mutated as part of the elimination
+ * @brief Representation of an LU factorization
+ *
+ * The entries of P mean row i of the matrix was exchanged with row P[i].
+ */
+template <typename T, int n>
+struct LuFactorization {
+  tensor<int, n>  P;  ///< Row permutation indices due to partial pivoting
+  tensor<T, n, n> L;  ///< Lower triangular factor. Has ones on diagonal.
+  tensor<T, n, n> U;  ///< Upper triangular factor
+};
+
+/**
+ * @brief Compute LU factorization of a matrix with partial pivoting
+ *
+ * The convention followed is to place ones on the diagonal of the lower
+ * triangular factor.
+ * @param[in] A The matrix to factorize
+ * @return An LuFactorization object
+ * @see LuFactorization
  */
 template <typename T, int n>
-SERAC_HOST_DEVICE constexpr tensor<T, n> linear_solve(tensor<T, n, n> A, const tensor<T, n> b)
+SERAC_HOST_DEVICE constexpr LuFactorization<T, n> lu(const tensor<T, n, n>& A)
```

Comment:
I would prefer `LU_factorize` or `factorize_LU` or something to be more explicit.

Reply:
I generally prefer verbing function names as well, with the exception of well understood math operations (eg `exp(A)` instead of `exponentiate(A)`). The LU factorization is a border case in my mind: `lu(A)` is fairly obvious in the context of tensors, but it isn't universal notation. If this thing were sitting in a more obvious context, (say `linear_algebra::lu(A)`) I'd argue to keep what I have, but it's just in the `serac` namespace. I've changed to `factorize_lu` - does this work?